src/rdkafka_fetcher.c

```diff
@@ -638,6 +768,10 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
                                                 4 /*PartitionArrayCnt*/ + 4 +
                                                 2 + 8 + 4 /*inner header*/));
 
+        fetch_tags.TopicCnt = TopicArrayCnt;
+        fetch_tags.Topics   = rd_calloc(
+            TopicArrayCnt, sizeof(rd_kafkap_Fetch_reply_tags_Topic_t));
```

Comment:
We allocate the array only when tags are supported
```suggestion
        if (rd_kafka_buf_ApiVersion(request) >= 12) {
                has_fetch_tags = rd_true;
                rd_kafkap_Fetch_reply_tags_set_topic_cnt(&FetchTags,
                                                      TopicArrayCnt);
        }
```

Reply:
Updated